io/subsystem/src/main/java/org/wildfly/extension/io/WorkerAdd.java

```diff
@@ -189,7 +221,14 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
             }
         }
 
-        final WorkerService workerService = new WorkerService(builder.getMap());
+        OptionMap options = builder.getMap();
+        final ThreadsService threadsService = new ThreadsService(options.get(Options.WORKER_TASK_MAX_THREADS));
```

Comment:
Makes sense, but to make this work EJB subsystem will need to have reference to name of the worker. 
Which is probably not an issue.